src/AreaChart/AreaSeries/AreaSeries.tsx

```diff
@@ -116,140 +118,143 @@ interface AreaSeriesState {
-            yScale={yScale}
-            data={data}
-            width={width}
-            index={index}
-            hasArea={area !== null}
-            animated={animated}
-            interpolation={interpolation}
-            color={this.getColor.bind(this)}
-          />
-        )}
-        {area && (
-          <CloneElement<AreaProps>
-            element={area}
-            id={`${id}-area-${index}`}
-            xScale={xScale}
-            yScale={yScale}
-            data={data}
-            index={index}
-            animated={animated}
-            interpolation={interpolation}
-            color={this.getColor.bind(this)}
-          />
-        )}
-      </Fragment>
-    );
-  }
-
-  renderSymbols(data: ChartInternalShallowDataShape[], index = 0) {
-    const {
-      xScale,
-      yScale,
-      symbols,
-      id,
-      height,
-      width,
-      animated,
-      area
-    } = this.props;
-    const { activeValues } = this.state;
-
-    const visible = symbols !== null;
-    const activeSymbols =
-      (symbols && symbols.props.activeValues) || activeValues;
-
-    // Animations are only valid for Area
-    const isAnimated = area !== undefined && animated && !activeSymbols;
+  const onValueLeave = () => {
+    setAreaState(initialAreaState);
+  };
 
-    return (
-      <Fragment>
-        {visible && (
-          <CloneElement<PointSeriesProps>
-            element={symbols}
-            key={`point-series-${id}`}
-            id={id}
-            height={height}
-            width={width}
-            activeValues={activeSymbols}
-            xScale={xScale}
-            yScale={yScale}
-            index={index}
-            data={data}
-            animated={isAnimated}
-            color={() => this.getColor(data, index)}
-          />
-        )}
-      </Fragment>
-    );
-  }
+  const renderArea = useCallback(
+    (data: ChartInternalShallowDataShape[], index = 0) => {
+      return (
+        <Fragment>
+          {line && (
+            <CloneElement<LineProps>
+              element={line}
+              xScale={xScale}
+              yScale={yScale}
+              data={data}
+              width={width}
+              index={index}
+              hasArea={area !== null}
+              animated={animated}
+              interpolation={interpolation}
+              color={getColor}
+            />
+          )}
+          {area && (
+            <CloneElement<AreaProps>
+              element={area}
+              id={`${id}-area-${index}`}
+              xScale={xScale}
+              yScale={yScale}
+              data={data}
+              index={index}
+              animated={animated}
+              interpolation={interpolation}
+              color={getColor}
+            />
+          )}
+        </Fragment>
+      );
+    },
+    [line, xScale, yScale, data, width, animated, interpolation]
+  );
+
+  const renderSymbols = useCallback(
+    (data: ChartInternalShallowDataShape[], index = 0) => {
```

Comment:
This is called data and the arg is called data. Should be renamed.

Reply:
fixed